### ISSUE TITLE
`server` lacks proper Custom Environments support

### ISSUE THREAD

**Issue by yuri-sergiichuk** [SpineEventEngine/core-java#1291]
With the introduction of the Custom Environments (see PR [#539](https://github.com/SpineEventEngine/base/pull/539)), we're now able to create custom environment types as `Beta` or `Staging`.

The feature itself is nice, but there are still places in the framework that treat `Production` in a very special way that is not portable for the usage of custom environment types.

Let's consider we have an application that is deployed in Production and Staging environments. Now, let's glance at the following parts of the framework where we should reconsider the current behavior.

-----

`ServerEnvironment#close`: 

```java ServerEnvironment.java
    @Override
    public void close() throws Exception {
        tracerFactory.ifPresentForEnvironment(Production.class, AutoCloseable::close);
        transportFactory.ifPresentForEnvironment(Production.class, AutoCloseable::close);
        storageFactory.ifPresentForEnvironment(Production.class, AutoCloseable::close);
    }
```

The code above will just not free the resources for the Staging environment even though it is the desired behavior.

-----

`StateSubscriberMethod#contextOf`:

```java StateSubscriberMethod.java
    private BoundedContextName contextOf(Class<?> cls) {
        Model model = Model.inContextOf(cls);
        BoundedContextName name = model.contextName();
        if (Environment.instance().is(Production.class) && name.equals(assumingTests())) {
            _warn().log(
                    "The class `%s` belongs to the Bounded Context named `%s`," +
                    " which is used for testing. As such, it should not be used in production." +
                    " Please see the description of `%s` for instructions on" +
                    " annotating packages with names of Bounded Contexts of your application.",
                    cls.getName(), assumingTests().getValue(), BoundedContext.class.getName()
            );
        }
        return name;
    }
```

We'll end up deploying code without `@BoundedContext` annotation up to the Production environment and we'll not notice the issue while running on the Staging.

-----

`SystemSettings#defaults`:

```java SystemSettings.java
    public static SystemSettings defaults() {
        SystemSettings settings = new SystemSettings()
                .disableCommandLog()
                .enableAggregateQuerying()
                .forgetEvents();
        if (Environment.instance().is(Production.class)) {
            settings.enableParallelPosting();
        } else {
            settings.disableParallelPosting();
        }
        return settings;
    }
```

The system events will be posted in parallel only in Production, but that should be the desired behavior on Staging as well.

**Comment by yuri-sergiichuk** [comment]
For the `StateSubscriberMethod` and `SystemSettings`, we can invert the condition and test that the environment is **not** Tests.

But to properly free the resources in the `ServerEnvironment` we should come up with something else. Maybe we can just iterate over all the present environments and free the resources for all of them?

**Comment by armiol** [comment]
@yuri-sergiichuk we just need to perform actions regardless of the environment type. Maybe (just maybe) we could treat the `Tests` in a special way, but only when posting the system events — otherwise the debugging may get more difficult. 

Other than that we should just do exactly the same.

Thanks for spotting the bug.

**Comment by dmitrykuzmin** [comment]
Fixed with #1293.
